src/terrain/clients/iplant_groups.clj

```diff
@@ -78,6 +79,19 @@
     (create-folder client user name))
   nil)
 
+;;
+
+(defn- format-group [folder group]
+  (let [regex (re-pattern (str "^\\Q" folder ":"))]
+    (update-in (dissoc group :detail) [:name] (fn [s] (string/replace s regex "")))))
+
+(defn- get-groups* [folder format-fn client user lookup-fn]
+  (if (folder-exists? client user folder)
+    {:groups (mapv format-fn (:groups (lookup-fn folder)))}
+    {:groups []}))
```

Comment:
This is some code that was generalized and extracted from `get-collaborator-lists*`.